src/main/java/graphql/validation/schemawiring/ValidationSchemaWiring.java

```diff
@@ -34,15 +36,23 @@ public ValidationSchemaWiring(ValidationRules ruleCandidates) {
     public GraphQLFieldDefinition onField(SchemaDirectiveWiringEnvironment<GraphQLFieldDefinition> env) {
         GraphQLFieldsContainer fieldsContainer = env.getFieldsContainer();
         GraphQLFieldDefinition fieldDefinition = env.getFieldDefinition();
-
+        TargetedValidationRules rules = ruleCandidates.buildRulesFor(fieldDefinition, fieldsContainer);
+        if (rules.isEmpty()) {
+            return fieldDefinition;
+        }
+        if (! (fieldsContainer instanceof GraphQLObjectType)) {
+            // only object type fields can have data fetchers
+            return fieldDefinition;
+        }
```

Comment:
The new version of grpahql-java 22 does not allow setting a data fetcher on GraphQLFieldsContainer anymore since that's nonsensical for interfaces